docs/src/domains/troubleshoot.md

```diff
@@ -25,25 +25,118 @@ main-abcd123.abcdefgh1234567.eu.platformsh.site has address 192.0.2.1
 
 If it isn't, try the following steps:
 
-* Your DNS server might not be correctly configured or the old DNS records are still cached.
+- Your DNS server might not be correctly configured or the old DNS records are still cached.
   Try removing your local DNS cache.
-* Set your computer's DNS server to Google Public DNS (`8.8.8.8` and `8.8.4.4`)
+- Set your computer's DNS server to Google Public DNS (`8.8.8.8` and `8.8.4.4`)
   to see if the issue is with the DNS server you are using.
-* Run `ping www.{{< variable "YOUR_DOMAIN" >}}`.
+- Run `ping www.{{< variable "YOUR_DOMAIN" >}}`.
   If the result is different from what you got from running `host www.{{< variable "YOUR_DOMAIN" >}}`,
   you might need to remove your test settings.
 
 ## Verify SSL
 
-On the command line using macOS, Linux, or the Windows Subsystem for Linux, run the following command:
+For troubleshooting, feel free to use [the certificate checker tool](https://certcheck.pltfrm.sh/).
+This tool assists in finding out where your domain is pointing to and provides some generic guidance.
+It also assists when a CDN (Fastly, Cloudflare, ...) is used.
+It's good practice to check both the apex and the `www` domain to ensure that both point to your project.
+
+For more in-depth investigations, on the command line using macOS, Linux, or the Windows Subsystem for Linux, run the following command:
 
 ```bash
 curl -I -v  https://www.{{< variable "YOUR_DOMAIN" >}}
 ```
 
 The response should be long. Look for error messages.
 They're usually explicit enough.
-Often the problem is with a mismatch between the certificate and the domain name.
+Often the problem is caused by a mismatch between the certificate and the domain name or an expired [custom certificate](steps/tls.md).
+
+### Error provisioning certificates
```

Comment:
:clipboard: That part should fit a bit better to the rest of the page and it's hopefully clearer, but might need more rework to be shorter and clearer.